consensus/src/alerts/service.rs

```diff
@@ -0,0 +1,148 @@
+    handler: Handler<H, D, MK>,
+    io: IO<H, D, MK>,
+}
+
+impl<H: Hasher, D: Data, MK: MultiKeychain> Service<H, D, MK> {
+    pub fn new(
+        keychain: MK,
+        messages_for_network: Sender<(NetworkMessage<H, D, MK>, Recipient)>,
+        messages_from_network: Receiver<NetworkMessage<H, D, MK>>,
+        notifications_for_units: Sender<ForkingNotification<H, D, MK::Signature>>,
+        alerts_from_units: Receiver<Alert<H, D, MK::Signature>>,
+        config: AlertConfig,
+    ) -> Service<H, D, MK> {
+        let n_members = config.n_members;
+        let handler = Handler::new(keychain.clone(), config);
+        let (messages_for_rmc, messages_from_us) = mpsc::unbounded();
+        let (messages_for_us, messages_from_rmc) = mpsc::unbounded();
+
+        let io = IO {
+            messages_for_network,
+            messages_from_network,
+            notifications_for_units,
+            alerts_from_units,
+            rmc: ReliableMulticast::new(
+                messages_from_us,
+                messages_for_us,
+                keychain,
+                n_members,
+                DoublingDelayScheduler::new(time::Duration::from_millis(500)),
+            ),
+            messages_from_rmc,
+            messages_for_rmc,
+            alerter_index: handler.index(),
+        };
+
+        Service { handler, io }
+    }
+
+    fn handle_message_from_network(
+        &mut self,
+        message: AlertMessage<H, D, MK::Signature, MK::PartialMultisignature>,
+    ) {
+        match self.handler.on_message(message) {
+            Ok(Some(AlerterResponse::ForkAlert(alert, recipient))) => {
+                self.io.send_message_for_network(
+                    AlertMessage::ForkAlert(alert),
+                    recipient,
+                    &mut self.handler.exiting,
+                );
+            }
+            Ok(Some(AlerterResponse::AlertRequest(hash, peer))) => {
+                let message = AlertMessage::AlertRequest(self.handler.index(), hash);
+                self.io
+                    .send_message_for_network(message, peer, &mut self.handler.exiting);
+            }
+            Ok(Some(AlerterResponse::RmcMessage(message))) => {
+                if self.io.messages_for_rmc.unbounded_send(message).is_err() {
+                    warn!(target: "AlephBFT-alerter", "{:?} Channel with messages for rmc should be open", self.handler.index());
+                    self.handler.exiting = true;
+                }
+            }
+            Ok(Some(AlerterResponse::ForkResponse(maybe_notification, hash))) => {
+                self.io.rmc.start_rmc(hash);
+                if let Some(notification) = maybe_notification {
+                    self.io
+                        .send_notification_for_units(notification, &mut self.handler.exiting);
+                }
+            }
+            Ok(None) => {}
+            Err(error) => debug!(target: "AlephBFT-alerter", "{}", error),
```

Comment:
Since you are touching the code here please also do the `LOG_TARGET` thingy we do in `aleph-node`.